prusti-tests/tests/typecheck/fail/abstract-predicates/invalid-sig-1.rs

```diff
@@ -0,0 +1,22 @@
+use prusti_contracts::*;
+
+trait SomeTrait {
+    predicate! {
+        fn foo(&self) -> i32;
+    }
+}
+
+struct SomeStruct;
+
+#[refine_trait_spec]
+impl SomeTrait for SomeStruct {
+    predicate! {
+        fn foo(&self) -> i32 {
+            42 //~ ERROR: mismatched types
```

Comment:
Why this error?

Reply:
I created a UI test, the error message is better there.